unity-renderer/Assets/DCLServices/Lambdas/LambdasService.cs

```diff
@@ -0,0 +1,123 @@
+using Cysharp.Threading.Tasks;
+using DCL;
+using System;
+using System.Text;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.Pool;
+
+namespace DCLServices.Lambdas
+{
+    public class LambdasService : ILambdasService
+    {
+        private ICatalyst catalyst;
+
+        private Service<IWebRequestController> webRequestController;
+
+        public UniTask<(TResponse response, bool success)> Post<TResponse, TBody>(
+            string endPoint,
+            TBody postData,
+            int timeout = ILambdasService.DEFAULT_TIMEOUT,
+            int attemptsNumber = ILambdasService.DEFAULT_ATTEMPTS_NUMBER,
+            CancellationToken cancellationToken = default,
+            params (string paramName, string paramValue)[] urlEncodedParams)
+        {
+            var postDataJson = JsonUtility.ToJson(postData);
+            var url = GetUrl(endPoint, urlEncodedParams);
+            var wr = webRequestController.Ref.Post(url, postDataJson, requestAttemps: attemptsNumber, timeout: timeout);
+
+            return SendRequestAsync<TResponse>(wr, cancellationToken, endPoint, urlEncodedParams);
+        }
+
+        public UniTask<(TResponse response, bool success)> Get<TResponse>(
+            string endPoint,
+            int timeout = ILambdasService.DEFAULT_TIMEOUT,
+            int attemptsNumber = ILambdasService.DEFAULT_ATTEMPTS_NUMBER,
+            CancellationToken cancellationToken = default,
+            params (string paramName, string paramValue)[] urlEncodedParams)
+        {
+            var url = GetUrl(endPoint, urlEncodedParams);
+            var wr = webRequestController.Ref.Get(url, requestAttemps: attemptsNumber, timeout: timeout);
+
+            return SendRequestAsync<TResponse>(wr, cancellationToken, endPoint, urlEncodedParams);
+        }
+
+        private async UniTask<(TResponse response, bool success)> SendRequestAsync<TResponse>(
+            IWebRequestAsyncOperation webRequestAsyncOperation,
+            CancellationToken cancellationToken,
+            string endPoint,
+            (string paramName, string paramValue)[] urlEncodedParams)
+        {
+            await webRequestAsyncOperation.WithCancellation(cancellationToken);
+
+            if (!webRequestAsyncOperation.isSucceeded)
+                return (default, false);
+
+            return !TryParseResponse(endPoint, urlEncodedParams, webRequestAsyncOperation.webRequest.downloadHandler.text, out TResponse response) ? (default, false) : (response, true);
+        }
+
+        internal string GetUrl(string endPoint, (string paramName, string paramValue)[] urlEncodedParams)
+        {
+            var urlBuilder = GenericPool<StringBuilder>.Get();
+            urlBuilder.Clear();
+
+            urlBuilder.Append(catalyst.Lambdas2Url);
+            urlBuilder.Append('/');
+            urlBuilder.Append(endPoint);
+
+            if (urlEncodedParams.Length > 0)
+            {
+                urlBuilder.Append("?");
```

Comment:
Because it already happened in the past that we had inconsistency between different endpoints (some auto placing `?` some not), can we do some validation to check if the endPoint variable already ends with a `?`?

Reply:
Done